_posts/2018-07-20-exploring-state-management-in-react-container-components.md

```diff
@@ -0,0 +1,122 @@
+---
+layout: post
+title: Container Components&#58; Exploring State Management in React (Part I)
+date: 2018-07-25 00:00:00 -04:00
+categories: post
+desc: This one simple trick will turn your unruly React components into the obedient angels they were meant to be!
+permalink: /:categories/:year/:month/:day/:title/
+---
+
+>At what level of complexity will my React application require Redux?
+
+React developers have been asking this question for a long time, and answers still vary wildly. The truth is there is quite a bit you can do before needing to pull in Redux, and even then, _**Redux**_ isn't your only option! 
+
+Even the creator of Redux, [Dan Abramov](https://twitter.com/dan_abramov?lang=en) thinks that you might not need Redux (although, I think the spirit this statement applies to all 3rd-party libraries meant to help reduce complexity of state).
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">You Might Not Need Redux <a href="https://t.co/3zBPrbhFeL">https://t.co/3zBPrbhFeL</a></p>&mdash; Dan Abramov (@dan_abramov) <a href="https://twitter.com/dan_abramov/status/777983404914671616?ref_src=twsrc%5Etfw">September 19, 2016</a></blockquote>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+In this series, we'll explore a few different patterns you can introduce to your code base before reaching for a 3rd party solution!
+
+## Container/Presenter Components
+
+This pattern separates what might be a single component into two (or three or four), a Container component(s) to maintain state and a Presenter component to render visual markup. 
```

Comment:
💡 I feel like it might be a bit too early to introduce the idea of multiple/nested Container components - you haven't even explained the pattern yet. I think this part would be better served at the end as a 'and here's how you can scale this pattern even further without Redux'.